### PR TITLE
feat: Validate form only on submit

### PR DESCRIPTION
**Pull request by rafenden**
GDS does not recommend validating forms using `onBlur` event, the PR will remove this behaviour:
https://design-system.service.gov.uk/components/error-message/

Thanks to @web-bert for pointing that out:
https://github.com/uktrade/data-hub-components/pull/97#issuecomment-524239998

### CODE REVIEW

File: src/forms/core/__tests__/useField.test.jsx [hunk 1]
```diff
@@ -29,38 +30,58 @@ const TestField = (props) => {
 }
 
 describe('useField', () => {
-  describe('when text is typed to the field', () => {
-    test('should update state value', () => {
-      const wrapper = mount(
+  describe('when field is mounted with `initialValue`', () => {
```
Comment (yeahfro) [inline src/forms/core/__tests__/useField.test.jsx]: 👍 these tests are looking quite tight

File: src/forms/core/useForm.js [hunk 2]
```diff
@@ -26,11 +30,11 @@ function useForm({ initialValues = {}, initialStep = 0, onSubmit = null } = {})
     }
 
     if (field && 'validate' in field) {
-      if (field.validate instanceof Function) {
+      if (typeof field.validate === 'function') {
```
Comment (yeahfro) [inline src/forms/core/useForm.js]: Based on using `isEmpty` from `lodash` I wonder if you should use `isFunction`

File: src/forms/core/useForm.js [hunk 3]
```diff
@@ -26,11 +30,11 @@ function useForm({ initialValues = {}, initialStep = 0, onSubmit = null } = {})
...
         return field.validate(value, name)
       }
 
-      if (field.validate instanceof Array) {
+      if (Array.isArray(field.validate)) {
```
Comment (yeahfro) [inline src/forms/core/useForm.js]: And `isArray`

File: src/forms/core/useForm.js (hunk 2 above)
Comment (rafenden) [reply src/forms/core/useForm.js]: As a rule of thumb, I prefer using built-in functions where possible to make the bundle smaller.

**Pull request merged**